### ISSUE TITLE
iOS11.2,After the alertViewController of the pop-up system, the HUD will shift up and not center

### ISSUE THREAD

**Issue by jniosdeveloper** [SVProgressHUD/SVProgressHUD#851]
Hello, I found a problem in ios 11.2,I don't know how to solve it, please See the code below:
```
UIAlertController *alertController = [UIAlertController alertControllerWithTitle:@"Okay?"
                                                                             message:@""
                                                                      preferredStyle:UIAlertControllerStyleAlert ];
    UIAlertAction *againAction = [UIAlertAction actionWithTitle:@"Yes" style:UIAlertActionStyleCancel handler:^(UIAlertAction * _Nonnull action) {
    }];
    [alertController addAction:againAction];
    
    UIAlertAction *cancelAction = [UIAlertAction actionWithTitle:@"NO" style:UIAlertActionStyleDefault handler:^(UIAlertAction * _Nonnull action) {
        
    }];
    [alertController addAction:cancelAction];

    [self presentViewController:alertController animated:YES completion:nil];
```
As soon as I pop up the alertViewController of the system, the HUD will shift upwards, and of course I only found it on iOS11, there's no problem with iOS10. How can I solve this problem?

**Comment by pchy9** [comment]
I found this problem too，have you soloved yet?

**Comment by tutumagi** [comment]
modify code in method `visibleKeyboardHeight`  in `SVProgressHUD.m`

```
- (CGFloat)visibleKeyboardHeight {
#if !defined(SV_APP_EXTENSIONS)
    UIWindow *keyboardWindow = nil;
    for (UIWindow *testWindow in UIApplication.sharedApplication.windows) {
        if(![testWindow.class isEqual:UIWindow.class]) {
            keyboardWindow = testWindow;
            break;
        }
    }
    
    for (__strong UIView *possibleKeyboard in keyboardWindow.subviews) {
        NSString *viewName = NSStringFromClass(possibleKeyboard.class);
        if([viewName hasPrefix:@"UI"]){
            if([viewName hasSuffix:@"PeripheralHostView"] || [viewName hasSuffix:@"Keyboard"]){
                return CGRectGetHeight(possibleKeyboard.bounds);
            } else if ([viewName hasSuffix:@"InputSetContainerView"]){
                for (__strong UIView *possibleKeyboardSubview in possibleKeyboard.subviews) {
                    viewName = NSStringFromClass(possibleKeyboardSubview.class);
                    if([viewName hasPrefix:@"UI"] && [viewName hasSuffix:@"InputSetHostView"]) {
//                       Modify this line to below
//                        return CGRectGetHeight(possibleKeyboardSubview.bounds);
                        CGRect visibleKeyboardFrame = CGRectIntersection(kScreenBounds, possibleKeyboardSubview.frame);
                        return CGRectGetHeight(visibleKeyboardFrame);
//                       end modify
                    }
                }
            }
        }
    }
#endif
    return 0;
}
```

**Comment by rpendleton** [comment]
This should be fixed in the latest release now that #846 has been merged in, right?

**Comment by honkmaster** [comment]
Indeed, this is fixed.
